### PR TITLE
Auto Update Fix

### PR DESCRIPTION
**Pull request by MDrakos**
https://www.pivotaltracker.com/story/show/179513231

### CODE REVIEW

**Review by mdrohmann — APPROVED** [review]
LGTM

File: cmd/state-installer/installer.go [hunk 1]
```diff
@@ -116,6 +116,11 @@ func run(out output.Outputer, installPath, sessionToken string, updateTag *strin
 		}
 	}
 
+	err = cfg.Set(installation.ConfigKeyInstallPath, installPath)
+	if err != nil {
+		logging.Error("Could not set install path in config: %v", errs.JoinMessage(err))
+	}
```
Comment (Naatan) [inline cmd/state-installer/installer.go:122]: This isn't necessary. By definition people hitting our use-case will use the default install path.

File: cmd/state-transition-update/main.go [hunk 2]
```diff
@@ -157,3 +170,31 @@ func runDefault() (rerr error) {
...
+		"path": filepath.Join(newInstallpath, filepath.Base(exec)),
+	}
+
+	fileBytes := box.Bytes(boxFile)
+	fileStr, err := strutils.ParseTemplate(string(fileBytes), tplParams)
+	if err != nil {
+		return errs.Wrap(err, "Could not parse %s template", boxFile)
+	}
+
+	if err = ioutil.WriteFile(exec, []byte(fileStr), 0755); err != nil {
```
Comment (Naatan) [inline cmd/state-transition-update/main.go:195]: I think for Linux/Mac we'd also have to mark the file executable.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: cmd/state-transition-update/main.go (hunk 2 above)
Comment (MDrakos) [reply cmd/state-transition-update/main.go:198]: Does the `0755` not cover this? The code here is very similar to the executor code [here](https://github.com/ActiveState/cli/blob/master/pkg/platform/runtime/executor/executor.go#L163)

Comment (Naatan) [reply cmd/state-transition-update/main.go:198]: I suppose it should. If you tested it and it worked then I'd say we're good to go.

File: assets/state/state.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+{{.path}} "$@"
```
Comment (Naatan) [inline assets/state/state.sh:3]:
```suggestion
"{{.path}}" "$@"
```

I think we'd need this on macOS/Linux as well, for paths with spaces (however unlikely that might be on those platforms).

**Review by Naatan — CHANGES_REQUESTED** [review]

File: cmd/state-transition-update/main.go (hunk 2 above)
Comment (MDrakos) [reply cmd/state-transition-update/main.go:198]: I tested this code when it was in the installer and it worked on macOS. Once this is in place we should put it through and end to end test.

**Review by Naatan — APPROVED** [review]

**Pull request merged**